v2/internal/genericarmclient/generic_client.go

```diff
@@ -66,7 +66,7 @@ func NewGenericClient(
 
 	ua := options.UserAgent
 	if ua == "" {
-		ua = userAgent
+		ua = DefaultUserAgent
```

Comment:
This will only be used if the code that calls `NewGenericClient()` doesn't specify a user agent in the `GenericClientOptions` struct passed in. I don't think that's ideal, as the configured setting wouldn't be included.

I haven't checked where `GenericClientIOptions.UserAgent` is set, but I'm wondering if the value used there should be used as a _part_ of the UA, not the whole thing.